modules/dnn/src/layers/nary_eltwise_layers.cpp

```diff
@@ -7,6 +7,7 @@
 #include "../op_cuda.hpp"
```

Comment:
1. We should not propagate changes between branches which massively breaks CI builds: #24954 (we still have lack of OpenCL testing on GHA)
2. This file is diverged too much between 5.x and 4.x (even before recent refactoring which makes things worse) - need to properly backport fixes from 5.x to 4.x to minimize difference.

Update of this file in this PR is not correct, some changes below 4.x..5.x are just lost.

5.x vs old 4.x:

```.diff
$ git diff 5.x-merge-4.x-origin..5.x -- modules/dnn/src/layers/nary_eltwise_layers.cpp
diff --git a/modules/dnn/src/layers/nary_eltwise_layers.cpp b/modules/dnn/src/layers/nary_eltwise_layers.cpp
index b22eb5bbf0..5170a45cd1 100644
--- a/modules/dnn/src/layers/nary_eltwise_layers.cpp
+++ b/modules/dnn/src/layers/nary_eltwise_layers.cpp
@@ -195,7 +195,7 @@ public:
         // since we'd need proper values of steps to check continuity.
         // this loop is probably the most tricky part
         // in the whole implementation of broadcasting.
-        j = max_ndims-1;
+        j = max_ndims > 0 ? max_ndims-1 : 0;
         for (i = j - 1; i >= 0; i--) {
             bool all_contiguous = true, all_scalars = true, all_consistent = true;
             for(k = 0; k < narrays; k++) {
@@ -225,6 +225,8 @@ public:
             for (k = 0; k < narrays; k++)
                 step[k][i] = shape[k][i] == 1 ? 0 : step[k][i];
         }
+        if (max_ndims == 0)
+            i = 0;
         for (; i >= 0; i--) {
             for (k = 0; k < narrays; k++) {
                 step[k][i] = 0;
@@ -252,13 +254,26 @@ public:
             char* data, const size_t* step,
             const Functor& op)
     {
-        assert(ndims >= 2);
-        size_t dp1 = step1[ndims-1]/sizeof(T);
-        size_t dp2 = step2[ndims-1]/sizeof(T);
-        size_t dp = step[ndims-1]/sizeof(T);
-        int k, n1 = shape[ndims-1], n2 = shape[ndims-2];
+        size_t dp1 = 0, dp2 = 0, dp = 0;
+        int k, n1 = 1, n2 = 1;
+        size_t inplane_step1 = 0, inplane_step2 = 0, inplane_step = 0;
         size_t plane_idx, nplanes = 1;
-        for (k = 0; k < ndims-2; k++) nplanes *= shape[k];
+
+        if (ndims >= 1) {
+            dp1 = step1[ndims-1]/sizeof(T);
+            dp2 = step2[ndims-1]/sizeof(T);
+            dp = step[ndims-1]/sizeof(T);
+            n1 = shape[ndims-1];
+
+            if (ndims >= 2) {
+                inplane_step1 = step1[ndims-2];
+                inplane_step2 = step2[ndims-2];
+                inplane_step = step[ndims-2];
+                n2 = shape[ndims-2];
+
+                for (k = 0; k < ndims-2; k++) nplanes *= shape[k];
+            }
+        }
 
         for (plane_idx = 0; plane_idx < nplanes; plane_idx++) {
             const char* ptr1_ = data1;
@@ -273,9 +288,9 @@ public:
                 ptr_ += i_k*step[k];
                 idx = next_idx;
             }
-            for (int i2 = 0; i2 < n2; i2++, ptr1_ += step1[ndims-2],
-                                            ptr2_ += step2[ndims-2],
-                                            ptr_ += step[ndims-2])
+            for (int i2 = 0; i2 < n2; i2++, ptr1_ += inplane_step1,
+                                            ptr2_ += inplane_step2,
+                                            ptr_ += inplane_step)
             {
                 const T* ptr1 = (const T*)ptr1_;
                 const T* ptr2 = (const T*)ptr2_;
@@ -316,31 +331,20 @@ public:
 
         int max_ndims = std::max(a.dims, std::max(b.dims, out.dims));
 
-        // buf holds the folllowing for a, b & output:
-        //  * orig_shapes, shapes (result_shape), orig_steps, steps (result_step), 3*4 elements in total
-        //  * shape_buf & step_buf, 3*2*max_ndims elements in total
-        //  * all_ndims, 3*1 elements in total
-        //  * all_type_sizes, 3*1 elements in total
-        AutoBuffer<size_t> buf(3 * (2 * max_ndims + 6));
-
-        int** orig_shapes = (int**)(buf.data());
-        int** shapes = orig_shapes + 3;
-        size_t** orig_steps = (size_t**)(shapes + 3);
-        size_t** steps = orig_steps + 3;
-
-        int* shape_buf = (int*)(steps + 3);
-        size_t* step_buf = (size_t*)(shape_buf + 3 * max_ndims);
-
-        int* all_ndims = (int*)(step_buf + 3 * max_ndims);
-        size_t* all_type_sizes = (size_t*)(all_ndims + 3);
+        const int* orig_shapes[3];
+        int shapes_[3][CV_MAX_DIM];
+        int* shapes[] = {shapes_[0], shapes_[1], shapes_[2]};
+        const size_t* orig_steps[3];
+        size_t steps_[3][CV_MAX_DIM];
+        size_t* steps[] = {steps_[0], steps_[1], steps_[2]};
+        int all_ndims[3];
+        size_t all_type_sizes[3];
 
         // assign orig_shapes, shapes, orig_steps, steps, all_ndims, all_type_sizes
         for (int i = 0; i < 3; i++)
         {
-            orig_shapes[i] = (int*)(i == 0 ? out_shape : in_shape[i-1]);
+            orig_shapes[i] = (const int*)(i == 0 ? out_shape : in_shape[i-1]);
             orig_steps[i] = (size_t*)(i == 0 ? out_step : in_step[i-1]);
-            shapes[i] = shape_buf + i * max_ndims;
-            steps[i] = step_buf + i * max_ndims;
             all_ndims[i] = i == 0 ? out_ndims : in_ndims[i-1];
             all_type_sizes[i] = sizeof(T);
         }
```

These changes are mostly from #23473 by @vpisarev

3. No idea why tests doesn't catch that broken code - perhaps there is lack of tests for added code.